### DISCUSSION TITLE
UploadFile to bytes

### DISCUSSION THREAD

**Discussion by ingcrengifo** [fastapi/fastapi#7610] [Questions] [answered]
### First check

* [X] I used the GitHub search to find a similar issue and didn't find it.
* [X] I searched the FastAPI documentation, with the integrated search.
* [X] I already searched in Google "How to X in FastAPI" and didn't find any information.

### Description

Hello, I have the next function
```
async def add_data (files: List [UploadFile] = File (...), params: str = Form (...)):
```

Since this function I sent to send a message to a sengrid, then I do:
```
"for file in files:"
```
but when I do:
```
attachment.file_content = FileContent (file)
```
I got an error, I could ask for the file on base64 but
```
encoded = base64.b64decode (file)
```
it doesn't work, thank you if you can guide me

**Comment by phy25** [comment]
What error do you get?

**Comment by ingcrengifo** [comment]
```
TypeError: Object of type bytes is not JSON serializable
```

Now I have changed the code to this, I have made several attempts but all fail.

```
    for file in files:
        data = await file.read()
        # rv = data.encode('utf-8')
        rv = base64.b64encode(data)
        print(rv, "rvrvrvrvr")
        attachment = Attachment()
        attachment.file_content = FileContent(rv)
        attachment.file_type = FileType(file.content_type)
        attachment.file_name = FileName(file.filename)
        attachment.content_id = ContentId('ejemplo')
        message.attachment = attachment
```

**Comment by ingcrengifo** [comment, marked as answer]
thanks, i just made it work:

![image](https://user-images.githubusercontent.com/12200313/81996459-adf03600-9612-11ea-91db-b50833b48c3c.png)

**Comment by tiangolo** [comment]
Thanks for the help here @phy25 ! :clap: :bow: 

Thanks for reporting back and closing the issue @INGCRENGIFO  :+1:

**Comment by Qleoz12** [comment]
> thanks, i just made it work:
> 
> ![image](https://user-images.githubusercontent.com/12200313/81996459-adf03600-9612-11ea-91db-b50833b48c3c.png)

hey do you have a sample code from this?
